assets/themes/bootstrap/resources/bootstrap/js/render_jobad_list_filter.js

```diff
@@ -27,120 +27,92 @@ function tableLoad() {
+                showHeader: 0,
+                columns: [{
+                    title: "",
+                    field: "title",
+                    class: "spanLink",
+                    formatter: titleFormatter
+                }, {
+                    title: "",
+                    field: "published",
+                    class: "spanLink",
+                    formatter: publishedFormatter
+                }]
+            });
+            /**
+             * Greps data from json_data, for table.
+             * @param {json_data} item array from json_data;
+             * @return {array} elements which tags are true to filtering tags, and will be shown in table;
+             */
             grepFunc = function (item) {
-            function hasTag(element, index, array) {
-                if (filterTag.length == 2){
-                    return element == filterTag[0],filterTag[1];
-                }else if (filterTag.length == 1){
-                    return element == filterTag[0];
-                }else if (filterTag.length == 3){
-                    return element == filterTag[0],filterTag[1],filterTag[2];
-                }else if (filterTag.length == 4){
-                    return element == filterTag[0],filterTag[1],filterTag[2],filterTag[3];
-                }else if (filterTag.length == 5){
-                    return element == filterTag[0],filterTag[1],filterTag[2],filterTag[3],filterTag[4];
+                /**
+                 * Checks if element has filtered tags.
+                 * @param {array} item.tags array with tags;
+                 * @return {boolean} true if every tag to filter are met in tags of element;
+                 */
+                function hasTag(element) {
+                    return element == filterTag;
                 }
-            }
+
                 return item.tags.some(hasTag);
             };
+            /**
+             * Loading table with filtered by default data.
+             */
+            tableLoad();
+        });
+        /**
+         * If requested URL has a #hash part, then we click on element which id=hash.
+         */
+        var hash = window.location.hash.substr(1);
+        if (!(hash == null)) {
+            setTimeout(function () {
+                document.getElementById(hash).click();
+            }, 100);
         }
-//table reloads with new, filtered data.
-    tableLoad();
-    });
-} catch(err) {
-console.log(err);
-}
+        /**
+         * Click on row = click to open detailed view
+         */
+        $("#table").on("click", "tr", function () {
+            $(this).find(".detail-icon").trigger("click");
+        });
+    } catch (err) {
+        console.log(err);
+    }
 });
-
-//Formatter for detailed view(Full view of entry row). 
+/**
+ * Formating data from json_data for detailed view of every row.
+ * for every key:value dictionaries of row it decorates and pushes strings to html.
+ */
 function detailFormatter(index, row) {
```

Comment:
describe every argument.
describe the result

It seems "index" unused - drop it ?